networkx/algorithms/approximation/broadcasting.py

```diff
@@ -0,0 +1,100 @@
+import networkx as nx
+from networkx import NetworkXError
+from networkx.utils import not_implemented_for
+
+__all__ = [
+    "tree_broadcast_time",
+]
+
+
+def _get_max_broadcast_value(G, U, v, values):
+    adj = sorted(set(G.neighbors(v)) & U, key=lambda u: values[u], reverse=True)
+    return max(values[u] + i for i, u in enumerate(adj, start=1))
```

Comment:
Here is my understanding behind why this works. 
The indices are there to "simulate" a sequence of broadcast calls. Since we know there is only one unique path between any two vertices in a tree, it makes sense to sort the adjacent vertices in descending order, and broadcast in that order. That way, once I have broadcast to a vertex with larger label (which actually represents the broadcast time of the sub-tree), it can start spreading calls to its child nodes, which makes the whole process faster.
We take the maximum of this sum, since this is what the total broadcast time from that vertex is going to be. (So both the number of adjacent vertices, and what their labels are affect the final broadcast time from that vertex).

Reply:
Consider the base case of a star graph. If we take the centre vertex, then its broadcast time is the number of adjacent vertices to it (since all the other labels will be zero). And suppose you took one of its neighbours and added say 3 child nodes (which means it now has label 3). Then you would want to broadcast to that neighbour first, so that it can pass begin passing the message as-well.